dll/win32/shlwapi/autocomp.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+ * PROJECT:     ReactOS shlwapi
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Implement SHAutoComplete
+ * COPYRIGHT:   Copyright 2020 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+#include <windef.h>
+#include <winreg.h>
+#include <shldisp.h>
+#include <shlguid.h>
+#define NO_SHLWAPI_STREAM
+#include <shlwapi.h>
+#include <atlbase.h>
+#include <atlcom.h>
+#include <atlstr.h>
+#include <atlsimpcoll.h>
+#include <strsafe.h>
+#include <wine/debug.h>
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+class CAutoCompleteEnumString :
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IEnumString
+{
+public:
+    CAutoCompleteEnumString(IAutoComplete2 *pAC2, DWORD dwSHACF, HWND hwndEdit)
+        : m_istr(0), m_hwndEdit(hwndEdit), m_dwSHACF(dwSHACF)
+    {
+        Reset();
+    }
+
+    CAutoCompleteEnumString(const CAutoCompleteEnumString& another)
+        : m_istr(another.m_istr)
+        , m_hwndEdit(another.m_hwndEdit)
+        , m_dwSHACF(another.m_dwSHACF)
+        , m_strs(another.m_strs)
```

Comment:
This is used for `Clone`?
Does that also need to be bound to the same edit control?

Reply:
Fixed to NULL.